src/jit/gtlist.h

```diff
@@ -97,6 +97,10 @@ GTNODE(SIMD_CHK         , GenTreeBoundsChk   ,0,GTK_SPECIAL|GTK_NOVALUE)// Compa
                                                                         // does the compare, so that it can be more easily optimized.  But that involves generating qmarks at import time...
 #endif // FEATURE_SIMD
 
+#ifdef FEATURE_HW_INTRINSICS
+GTNODE(HW_INTRINSIC_CHK  , GenTreeBoundsChk   ,0,GTK_SPECIAL|GTK_NOVALUE)// Compare whether an imm8 argument is in the valid range, and throw ArgumentOutOfRangeException if not.
```

Comment:
It appears that this new oper has the same functionality as the existing `SIMD_CHK`. If that's the case then the existing oper should be used instead of adding a new one.

Reply:
> same functionality as the existing SIMD_CHK

But the semantic is different, `SIMD_CHK` is used to check array bounds.

Reply:
No, `GT_ARR_BOUNDS_CHECK` does that. Well, sort of. They all do the same thing - check if an integer is `>= 0` and `< gtArrLen` where `gtArrLen` can be any arbitrary tree (at least in theory). The difference is that some JIT parts (anything that has to do with redundant range check elimination) pay attention only to `GT_ARR_BOUNDS_CHECK`.

Reply:
> Even the existence of GT_SIMD_CHK is questionable, there's a good chance that GT_ARR_BOUNDS_CHECK could have been used for that as well.

It is possible that they could be merged, but at the time we added `GT_SIMD_CHK` there was code that relied on the certain expected code patterns for `GT_ARR_BOUND_CHECK`, and it was felt that it would be less risky to avoid overloading it. It may be reasonable to merge these but I don't see it as critical. It may make more sense to merge all three to a `GT_RANGE_CHECK` in future, if we determine that it is more efficient.

Reply:
Yes, merging with GT_ARR_BOUNDS_CHECK is a separate issue. But we shouldn't be adding a 3rd range check node without a good reason.

Reply:
I concede your point. These should probably be merged, though I am not 100% sure that is without risk, and I really don't want to hold this up for something that might lead to complications.